public/app/plugins/datasource/loki/LogContextProvider.ts

```diff
@@ -75,15 +80,17 @@ export class LogContextProvider {
   getLogRowContextQuery = async (
     row: LogRowModel,
     options?: LogRowContextOptions,
-    origQuery?: LokiQuery
+    origQuery?: LokiQuery,
+    forceApplyFilters?: boolean
   ): Promise<LokiQuery> => {
     // FIXME: This is a hack to make sure that the context query is created with
     // the correct set of filters. The whole `appliedContextFilters` property
     // should be revisted.
-    const cachedFilters = this.appliedContextFilters;
-    this.appliedContextFilters = [];
-    const { query } = await this.getQueryAndRange(row, options, origQuery);
-    this.appliedContextFilters = cachedFilters;
+    const { query } = await this.getQueryAndRange(row, options, origQuery, forceApplyFilters);
+
+    if (forceApplyFilters) {
+      this.appliedContextFilters = [];
```

Comment:
Also here, we need to add comment that explains that if `forceApplyFilters` is called, we need to reset filters after.